### PR TITLE
Final Document PR

### PR DESCRIPTION
**Pull request by weidious**
This PR is to upload an event trigger document following the new documentation format.

### CODE REVIEW

File: docs/eventTrigger.rst [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+*****
+Limitation
+*****
+Since our design purpose is to decouple the trigger condition from the action to take, currently there is a limitation on deserializing record. Although Kafka provides the ability to publish and subscribe to streams of records on custom serializer and deserializer. What we have right now is limited to Kafka build in String deserializer only. We are planning to enhance the flexibility on users to upload JAR with their own custom deserialize function in the near future. 
+
+*****
+Resource Referance
+*****
+- `Event Trigger Design Proposal <https://docs.google.com/document/d/1KxT17D8nvPvB-yTVZZEKuLMDp_TjdN1L6vagcZ4ypRI/edit>`_
```
Comment (chengren311) [inline docs/eventTrigger.rst]: please remove these internal resources as this document is mostly for providing know-hows and instructions on the feature for OSS users.

File: docs/eventTrigger.rst [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+Flow Trigger Dependency Plugin
+==================================
+*****
+Event Based Trigger
+*****
+
+
+
+Currently there are only few ways to launch jobs in Azkaban including schedules and API. However, they are limited because sometimes jobs need to be executed automatically on demand. Event trigger is a new feature introduced by Azkaban. It defines a new paradigm of triggering flows - triggering a flow on event arrival. This concept enables users to define events that the flow depends on. Once all of the dependencies become ready, a workflow will be triggered.
```
[the remark this answers is not in the document]
Comment (weidious) [reply docs/eventTrigger.rst]: ToDo note is added into .rst file already for the future needs.

**Comment by kunkun-tang** [comment]
Since trigger service is only applicable to web server, so non-solo server configs should be the same as solo server. @inramana

File: docs/eventTrigger.rst [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+
+..
+   Todo:: Link to the data trigger documentation if available
+Currently there are only few ways to launch jobs in Azkaban including schedules and API. However, they are limited because sometimes jobs need to be executed automatically on demand. Event trigger is a new feature introduced by Azkaban. It defines a new paradigm of triggering flows - triggering a flow on event arrival. This concept enables users to define events that the flow depends on. Once all of the dependencies become ready, a workflow will be triggered. 
+
+Apache Kafka is a Publish & Subscribe data streaming system. By utilizing Kafka, we do the regular expression match on the Kafka event payload. With the contain-a logic matching, a dependency will be marked as satisfied only if the whole payload contains the Regex pattern that user predefines.
+
+*****
+Getting started with Deploying Event Trigger on the Azkaban
```
Comment (chengren311) [inline docs/eventTrigger.rst]: first letter uppercase

File: docs/eventTrigger.rst [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+These are all standard Gradle commands. Please look at Gradle documentation for more info.
+
+
+
+Server Configuration
+########
+The gradlew commands help you to build the fat JAR. After that, you need to specify the plugin.dir within ``conf``. Take solo-server for example, override the ``azkaban.dependency.plugin.dir`` property for runtime parameters inside the ``azkaban.properties`` file under the solo-server ``conf`` directory.
+This property needs to set to contain the location where you put your Event-Trigger JAR file. 
+
+Data Base Configuration (Optional)
```
Comment (chengren311) [inline docs/eventTrigger.rst]: database

Comment (chengren311) [reply docs/eventTrigger.rst]: why optional

File: docs/eventTrigger.rst [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+########
+The following 4 properties can be defined in ``conf/azkaban.private.properties``. for solo-server based on the use case.
```
Comment (chengren311) [inline docs/eventTrigger.rst]: can be or has to be?

File: docs/eventTrigger.rst [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+Event trigger is part of flow definition and each flow can only have one event trigger at most. 
+Defining an event trigger is supported via Hadoop DSL.
+The trigger needs to be configurated within the flow file along with the project zip that users upload.
+Event trigger is composed of a list of event dependencies, max wait time and schedule.
+Take the following figure as example:
+
+.. image:: figures/TriggerExample.png
+
+- **Max Wait Time**: How long the trigger will wait for all dependencies to be available before cancelling it.
+- **Trigger.schedule**: The schedule to perform this workflow on the regular basis. We use the cron time format here to specify, creating a trigger followed by the project workflow every 2 minutes
```
Comment (chengren311) [inline docs/eventTrigger.rst]: duplicate trigger.schedule?

Comment (chengren311) [reply docs/eventTrigger.rst]: the schedule to perform availability check?

File: docs/eventTrigger.rst [hunk 7]
```diff
@@ -0,0 +1,156 @@
+.. _EventBasedTrigger:
+
+
+Flow Trigger Dependency Plugin
+==================================
+*****
+Event Based Trigger
```
Comment (chengren311) [inline docs/eventTrigger.rst]: Kafka Event based trigger

File: docs/eventTrigger.rst [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+*****
+
+
+..
+   Todo:: Link to the data trigger documentation if available
+Currently there are only few ways to launch jobs in Azkaban including schedules and API. However, they are limited because sometimes jobs need to be executed automatically on demand. Event trigger is a new feature introduced by Azkaban. It defines a new paradigm of triggering flows - triggering a flow on event arrival. This concept enables users to define events that the flow depends on. Once all of the dependencies become ready, a workflow will be triggered.
```
Comment (chengren311) [inline docs/eventTrigger.rst]: Azkaban supports launching flows via scheduling it or Ajax API.

Comment (chengren311) [reply docs/eventTrigger.rst]: triggering a flow on event arrival -> triggering a flow on kafka event arrival

Comment (chengren311) [reply docs/eventTrigger.rst]: This concept enables users to define events that the flow depends on. Once all of the events become ready, a workflow will be triggered.

**Pull request merged**